### PR TITLE
New package: setzer-0.3.2

### PR DESCRIPTION
**Pull request by Bleznudd**
Adding [setzer](https://github.com/cvfosammmm/Setzer) v0.3.2

### CODE REVIEW

File: srcpkgs/setzer/template
Comment (PaperMountainStudio) [inline srcpkgs/setzer/template:7]: remove this comment

File: srcpkgs/setzer/template
Comment (PaperMountainStudio) [inline srcpkgs/setzer/template:9]: this one too

File: srcpkgs/setzer/template
Comment (PaperMountainStudio) [inline srcpkgs/setzer/template:10]: meson dependency is detected automatically, you don't need to specify it

**Review by PaperMountainStudio — CHANGES_REQUESTED** [review]

**Comment by PaperMountainStudio** [comment]
commit message should be "New package: $pkgname-$version."

File: srcpkgs/setzer/template
Comment (PaperMountainStudio) [inline srcpkgs/setzer/template:15]: maybe https://www.cvfosammmm.org/setzer/?

File: srcpkgs/setzer/template
Comment (PaperMountainStudio) [inline srcpkgs/setzer/template:5]: why is this here? I tried aarch64 and armv7l-musl and they build fine (didn't test if they work correctly).

File: srcpkgs/setzer/template
Comment (PaperMountainStudio) [reply srcpkgs/setzer/template:10]: python3-gobject-devel should be moved to makedepends. Are python3-xdg, xdg-utils and gobject-introspection really needed here? It builds fine without them.

File: srcpkgs/setzer/template
Comment (Bleznudd) [reply srcpkgs/setzer/template]: done

File: srcpkgs/setzer/template
Comment (Bleznudd) [reply srcpkgs/setzer/template]: done

File: srcpkgs/setzer/template
Comment (Bleznudd) [reply srcpkgs/setzer/template]: done

File: srcpkgs/setzer/template
Comment (Bleznudd) [reply srcpkgs/setzer/template]: Meson was removed, about python3-xdg and the other I didn't test building without them, the Setzer README clearly says:

> Run the following command to install prerequisite Debian packages:
> ```
> apt-get install python3-gi gir1.2-gtk-3.0 gir1.2-gtksource-4 gir1.2-gspell-1 gir1.2-pango-1.0 gir1.2-poppler-0.18 gir1.2-webkit2-4.0 python3-xdg gettext xdg-utils
> ```

Are we sure they aren't nedeed? If so, I'll remove them

File: srcpkgs/setzer/template
Comment (Bleznudd) [reply srcpkgs/setzer/template:5]: I can't test them neither. What should we write under archs if it build fine but we can't test if it actually works?

Comment (PaperMountainStudio) [reply srcpkgs/setzer/template:5]: You can test them for example using the command `xbps-src -a aarch64 pkg setzer`, but I don't see why it shouldn't build fine. Just remove this line and if travis fails, you can add it again with the proper architectures.

File: srcpkgs/setzer/template
Comment (PaperMountainStudio) [reply srcpkgs/setzer/template]: The README doesn't differentiate between runtime dependencies and build dependencies. These packages are not needed for the build. Move them to `depends=` if they are needed for runtime, you will have to test this.

File: srcpkgs/setzer/template
Comment (Bleznudd) [reply srcpkgs/setzer/template]: Travis didn't fail, it should be ok now

**Comment by ericonr** [comment]
This is missing several dependencies. After installing, I got:

```
Traceback (most recent call last):
  File "/bin/setzer", line 19, in <module>
    gi.require_version('Gtk', '3.0')
AttributeError: module 'gi' has no attribute 'require_version'
```

After installing `python3-gobject`, I now have:

```
Traceback (most recent call last):
  File "/bin/setzer", line 26, in <module>
    from setzer.workspace.workspace import Workspace
  File "/usr/lib/python3.8/site-packages/setzer/workspace/workspace.py", line 22, in <module>
    from setzer.document.document import Document, LaTeXDocument, BibTeXDocument
  File "/usr/lib/python3.8/site-packages/setzer/document/document.py", line 25, in <module>
    import setzer.document.document_viewgtk as document_view
  File "/usr/lib/python3.8/site-packages/setzer/document/document_viewgtk.py", line 20, in <module>
    gi.require_version('GtkSource', '4')
  File "/usr/lib/python3.8/site-packages/gi/__init__.py", line 129, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace GtkSource not available
```

**Comment by Bleznudd** [comment]
Ok so `python3-gobject` is required, sorry for missing that.
About the second traceback: could it be resolved by installing `gtksourceview4`?

File: srcpkgs/setzer/template
Comment (ericonr) [reply srcpkgs/setzer/template]: For python programs, you want to watch the `do_build` output to be sure python isn't trying to get dependencies using `EasyInstall` stuff. If it is, the dependencies should go in `hostmakedepends`, since the host's python interpreter needs to be able to find them. These same packages, when necessary for runtime, should also go in `depends`.

**Comment by ericonr** [comment]
Yeah,

```diff
diff --git a/srcpkgs/setzer/template b/srcpkgs/setzer/template
index e0c57c8ba9..7516bd03fe 100644
--- a/srcpkgs/setzer/template
+++ b/srcpkgs/setzer/template
@@ -6,7 +6,7 @@ wrksrc=Setzer-${version}
 build_style=meson
 hostmakedepends="gettext"
 makedepends="python3-gobject-devel"
-depends="libpoppler89 texlive-core gspell python3-xdg"
+depends="libpoppler89 texlive-core gspell python3-xdg python3-gobject gtksourceview4"
 short_desc="Simple yet full-featured LaTeX editor written in Python with Gtk"
 maintainer="Bleznudd <bleznudd@pm.me>"
 license="GPL-3.0-or-later"
```

was enough. It could still be using stuff I already had installed, so it would be nice to test it more.

**Comment by ericonr** [comment]
Looks good, thanks! I will wait to merge this, since we are looking at updating poppler libraries.

**Comment by ahesford** [comment]
Yes, please wait until I merge #25126. Also, why does this have a dependency on `libpoppler89`? This is almost certainly wrong. Either it calls `dlopen` on (the wrong) versioned `libpoppler`, or it expects an unversioned symlink that will not be installed by `libpoppler89`, or `poppler-devel` should instead be listed in `makedepends`.

**Comment by ericonr** [comment]
I had it in the back of my mind, thanks for making a note about it.

File: srcpkgs/setzer/template
Comment (ericonr) [reply srcpkgs/setzer/template:9]: `virtual?tex`

Comment (Bleznudd) [reply srcpkgs/setzer/template]: >Theoretically, someone could use texlive-bin with this package and expect everything to work, correct?

I guess you're correct. Changes have been pushed

Comment (Bleznudd) [reply srcpkgs/setzer/template]: It seems that `virtual?tex` is giving problem on arm (`aarch64`, `armv7l`, `armv6l-musl` and `aarch64-musl`), travis log says:

```
=> texlive-20200406_4: installing host dependencies: pkg-config-0.29.2_2 perl-5.30.3_1 lua52-BitOp-1.0.2_3 texinfo-6.7_1 texlive-20200406_4 cross-i686-linux-musl-0.33_3 ...

=> texlive-20200406_4: installing target dependencies: cairo-devel-1.16.0_2 freetype-devel-2.10.2_1 gd-devel-2.3.0_2 graphite-devel-1.3.14_1 gmp-devel-6.2.0_1 harfbuzz-devel-2.7.2_2 icu-devel-67.1_1 libpaper-devel-1.1.28_1 libpng-devel-1.6.37_1 mpfr-devel-4.0.2_1 poppler-devel-20.09.0_1 pixman-devel-0.38.4_1 libteckit-devel-2.5.10_1 zlib-devel-1.2.11_3 zziplib-devel-0.13.71_1 libXaw-devel-1.0.13_2 ...

=> texlive-20200406_4: running do-fetch hook: 00-distfiles ...

=> texlive-20200406_4: fetching distfile 'texlive-20200406-source.tar.xz'...

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received

The build has been terminated
```

Any clue?

Comment (ericonr) [reply srcpkgs/setzer/template]: Unsynced mirror, it's trying to build `texlive` in CI to fulfill the dependency. Not a problem.

File: srcpkgs/setzer/template
Comment (ericonr) [inline srcpkgs/setzer/template:14]: Since you're going to have to push to retrigger CI, would you mind changing this to `distfiles="${homepage}/...`?

Comment (Bleznudd) [reply srcpkgs/setzer/template]: Sure

**Pull request merged**
